rust/frameworks/tw_ton_sdk/src/boc/boc_to_raw_boc.rs

```diff
@@ -96,7 +101,7 @@ fn build_and_verify_index(roots: &[CellArc]) -> HashMap<H256, IndexedCellRef> {
         }
     }
 
-    cells_by_hash
+    cells_by_hash.into_iter().collect()
```

Comment:
This still will be non-deterministic. That's because `HashMap` is randomly seeded:
https://doc.rust-lang.org/stable/std/collections/struct.HashMap.html
Look like we need to replace `HashMap<H256, IndexedCellRef>` with `BTreeMap<H256, IndexedCellRef>` everywhere in this file, and even add a type alias like `type CellsByHash = BTreeMap<H256, IndexedCellRef>`.

Reply:
You are right. A HashMap is unordered; however, the subsequent code does not iterate over it directly; instead, it converts it to a Vec and then sorts it.

https://github.com/trustwallet/wallet-core/blob/af67cae30f132ec139b72b4b93d2b336b84a8c49/rust/frameworks/tw_ton_sdk/src/boc/boc_to_raw_boc.rs#L30
So, I think it's deterministic.

I just want to keep the modifications minimal.

Reply:
It's also passed into 
https://github.com/trustwallet/wallet-core/blob/af67cae30f132ec139b72b4b93d2b336b84a8c49/rust/frameworks/tw_ton_sdk/src/boc/boc_to_raw_boc.rs#L41-L42

Reply:
> It's also passed into
> 
> https://github.com/trustwallet/wallet-core/blob/af67cae30f132ec139b72b4b93d2b336b84a8c49/rust/frameworks/tw_ton_sdk/src/boc/boc_to_raw_boc.rs#L41-L42

The scenarios you mentioned do not iterate over the HashMap.

Reply:
Ok I see, you're right, I didn't notice that.
We still have two options with pros and cons:
1. Use BTreeMap within `build_and_verify_index` only, and then convert it into HashMap.
Cons: Double memory allocation + iterate over all items within BTreeMap to insert them into HashMap
2. Use BTreeMap within `build_and_verify_index` and other functions too.
Cons: Searching for a Cell hash within BTreeMap has O(logn) complexity, while searching within HashMap has O(1).

I'd follow the second way tbh as it's more clear, and less overhead from my point of view

Reply:
I've made the requested adjustments to the code. Could you please review it again? The code has been formatted using `cargo fmt`.